bin/generate_commit

```diff
@@ -0,0 +1,169 @@
+    # For Glean pings, we copy the generic Glean schema into place later on.
+    mozilla-schema-generator generate-main-ping \
+        --mps-branch "$mps_branch_source" \
+        --out-dir ./telemetry
+
+    mozilla-schema-generator generate-common-pings \
+        --common-pings-config "$COMMON_PINGS_PATH" \
+        --mps-branch "$mps_branch_source" \
+        --out-dir ./telemetry
+
+    mozilla-schema-generator generate-glean-pings \
+        --mps-branch "$mps_branch_source" \
+        --out-dir .
+
+    # Remove all non-json schemas (e.g. parquet)
+    find . -not -name "*.schema.json" -type f -exec rm {} +
+
+    # Add metadata fields to all json schemas
+    # schema directory structure is enforced by regex at compile-time
+    # shellcheck disable=SC2044
+    for schema in $(find . -name "*.schema.json" -type f); do
+        metadata_merge metadata/ "$schema"
+    done
+
+    # Add transpiled BQ schemas
+    find . -type f -name "*.schema.json" | while read -r fname; do
+        # This schema is AWS-specific, fails transpilation, and should be ignored
+        if [[ $fname =~ metadata/sources ]] ; then
+            continue
+        fi
+        bq_out=${fname/schema.json/bq}
+        mkdir -p "$(dirname "$bq_out")"
+        jsonschema-transpiler \
+            --resolve drop \
+            --type bigquery \
+            --normalize-case \
+            --force-nullable \
+            --tuple-struct \
+                "$fname" > "$bq_out"
+    done
+
+    # Remove BigQuery schemas that are in the disallow list
+    find . -name '*.bq' | grep -f "$DISALLOWLIST" | xargs rm -f
+
+    # Copy aliased BQ schemas into place
+    alias_schemas "$ALIASES_PATH" .
+    popd
+}
+
+function _create_changelog() {
+    # Generate output for referencing to the changeset in the source branch.
+    # Useful references for --format and --pretty
+    # https://stackoverflow.com/questions/25563455/how-do-i-get-last-commit-date-from-git-repository
+    # https://stackoverflow.com/questions/1441010/the-shortest-possible-output-from-git-log-containing-author-and-date
+    # https://git-scm.com/docs/git-log/1.8.0#git-log---daterelativelocaldefaultisorfcshortraw
+    local mps_branch_source=${1:-master}
+    local mps_branch_publish=${2:-generated-schemas}
+    local start_date
+    # NOTES: Since this function is looking at commit dates (such as rebases),
+    # it's best to enforce squash/rebase commits onto the master branch. If this
+    # isn't enforced, it's possible to miss out on certain commits to the
+    # generated branch. Another solution is to generate a tag on master before
+    # the generator is run. However, the heuristic of using the latest commit
+    # date works well enough.
+    start_date=$(git log "${mps_branch_publish}" -1 --format=%cd --date=iso)
+    git log "${mps_branch_source}" \
+        --since="$start_date" \
+        --pretty=format:"%h%x09%cd%x09%s" \
+        --date=iso
+}
+
+function commit_schemas() {
```

Comment:
Did anything change when you moved these functions to the new file?

Reply:
The only thing that changed is that these now require path to the schema repository instead of relying on the absolute path within the container. They generated the same schemas.